orchestrator/ethereum_gravity/src/erc20_utils.rs

```diff
@@ -0,0 +1,75 @@
+use crate::types::EthClient;
+use ethers::prelude::*;
+use gravity_abi::erc20::ERC20;
+use gravity_utils::error::GravityError;
+use std::time::Duration;
+
+/// Checks if any given contract is approved to spend money from any given erc20 contract
+/// using any given address. What exactly this does can be hard to grok, essentially when
+/// you want contract A to be able to spend your erc20 contract funds you need to call 'approve'
+/// on the ERC20 contract with your own address and A's address so that in the future when you call
+/// contract A it can manipulate your ERC20 balances. This function checks if that has already been done.
+pub async fn check_erc20_approved(
+    erc20: Address,
+    target_contract: Address,
+    address: Address,
+    eth_client: EthClient,
+) -> Result<bool, GravityError> {
+    let erc20_contract = ERC20::new(erc20, eth_client.clone());
+    let contract_call = erc20_contract.allowance(address, target_contract);
+    let allowance = contract_call.call().await?;
+
+    // TODO(bolten): verify if this check is sufficient/correct
+    // Check if the allowance remaining is greater than half of a U256 - it's as good
+    // a test as any.
+    Ok(allowance > (U256::MAX.div_mod(2u32.into()).0))
+}
```

Comment:
I think this function would be better if expected amount of allowance was passed in as an argument

Reply:
Probably! This was one of the functions that was ported from web30. I can update this to take it as an argument, although we're still in the place of passing this mysterious constant in from the calling side.